src/axom/sidre/core/MFEMSidreDataCollection.hpp

```diff
@@ -259,6 +259,44 @@ class MFEMSidreDataCollection : public mfem::DataCollection
                      const std::string& buffer_name,
                      IndexType offset);
 
+   /// Register a mfem::QuadratureFunction in the Sidre DataStore.
+   /** This method is a shortcut for the call
+       `RegisterQField(field_name, qf, field_name, 0)`.
+    */
+   virtual void RegisterQField(const std::string &field_name, mfem::QuadratureFunction *qf)
+   {
+      RegisterQField(field_name, qf, field_name, 0);
+   }
+
+   /// Register a mfem::QuadratureFunction in the Sidre DataStore.
+   /** The registration procedure is as follows:
+       - if (@a qf's data is NULL), error out since this means there's no quadrature
+         space associated with it;
+       - else, if (DataStore has a named buffer @a buffer_name), replace @a qf's
+         data array with that named buffer plus the given @a offset care will be
+         taken to ensure the two have compatible sizes;
+       - else, use @a qf's data as external data associated with @a field_name
+         in the DataStore;
+       - register @a field_name in #qfield_map.
+       Both the @a field_name and @a buffer_name can contain a path prefix.
+       @note If @a field_name or @a buffer_name is empty, the method does
+       nothing.
+       @note If the QuadratureFunction pointer @a qf or it's QuadratureSpace
+       pointers are NULL, the method does nothing.
+    */
+   void RegisterQField(const std::string &field_name, mfem::QuadratureFunction *gf,
+                      const std::string &buffer_name,
+                      axom::sidre::IndexType offset);
+   
+   /// Returns the QuadratureSpace order useful for reconstructing a QuadratureFunction
+   int GetQFieldOrder(const std::string &field_name);
+
+   /// Returns the QuadratureSpace VDim useful for reconstructing a QuadratureFunction
+   int GetQFieldVDim(const std::string &field_name);
+
+   /// Returns a pointer to the raw data useful for reconstructing a QuadratureFunction
+   double* GetQFieldData(const std::string &field_name);
```

Comment:
Is the intent that these will be going away/moved once the reconstruction logic is finished?  I'd be hesitant to add them to the public interface of the class.